server/utils/firebase_algolia_server_code/functions/test/index.test.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Unit tests for index.js
+ * Uses dependency injection to mock algolia save/delete functions
+ * Uses sinon/chai to spy on calls and validate behavior
+ * compile with `npm run test`
+ * Having problems running this on your local? 
+ * - Download a firebase service key.
+ * - *DON'T* push this key to github please.  
+ */
+
+// Initialize the firebase test mocking environment.
+const test = require('firebase-functions-test')({
+  databaseURL: 'https://step9-2020-capstone.firebaseio.com',
+  storageBucket: 'step9-2020-capstone.appspot.com',
+  projectId: 'step9-2020-capstone',
+}, 'step9-2020-capstone.json');
+
+// Initialize chai, sinon, sinon-chai. 
+var assert = require('chai').assert;
```

Comment:
never use `var` - it's as evil as the regex or `text-align` lol. Try to use `const` as much as possible, and use `let` if you really have to change the variable's value later